packages/vscode-extension/src/debugging/ProxyDebugAdapter.ts

```diff
@@ -0,0 +1,299 @@
+            "**/vscode-extension/lib/**/*.js",
+            "**/ReactFabric-dev.js",
+            "**/ReactNativeRenderer-dev.js",
+            "**/node_modules/**/*",
+            "!**/node_modules/expo-router/**/*",
+          ],
+        },
+        {
+          suppressDebugStatusbar: true,
+          suppressDebugView: true,
+          suppressDebugToolbar: true,
+          suppressSaveBeforeStart: true,
+          parentSession: this.session,
+          consoleMode: vscode.DebugConsoleMode.MergeWithParent,
+          lifecycleManagedByParent: true,
+          compact: true,
+        }
+      );
+
+      if (!childSessionStarted) {
+        this.sendErrorResponse(
+          response,
+          { format: "Failed to attach debugger session", id: 1 },
+          undefined,
+          undefined,
+          ErrorDestination.User
+        );
+      }
+
+      assert(this.nodeDebugSession !== null);
+      this.sendResponse(response);
+    } finally {
+      didStartSessionHandler?.dispose();
+    }
+  }
+
+  protected continueRequest(
+    response: DebugProtocol.ContinueResponse,
+    args: DebugProtocol.ContinueArguments,
+    request?: DebugProtocol.Request
+  ): void {
+    if (!this.nodeDebugSession) {
+      return;
+    }
+    vscode.commands.executeCommand("workbench.action.debug.continue", undefined, {
+      sessionId: this.nodeDebugSession.id,
+    });
+  }
+
+  protected nextRequest(
+    response: DebugProtocol.NextResponse,
+    args: DebugProtocol.NextArguments,
+    request?: DebugProtocol.Request
+  ): void {
+    if (!this.nodeDebugSession) {
+      return;
+    }
+    vscode.commands.executeCommand("workbench.action.debug.stepOver", undefined, {
+      sessionId: this.nodeDebugSession.id,
+    });
+  }
+
+  protected async disconnectRequest(
+    response: DebugProtocol.DisconnectResponse,
+    args: DebugProtocol.DisconnectArguments,
+    request?: DebugProtocol.Request
+  ) {
+    this.terminate();
+    this.sendEvent(new Event("RNIDE_continued"));
+    this.sendResponse(response);
+  }
+
+  private terminate() {
+    if (this.terminated) {
+      return;
+    }
+    this.terminated = true;
+    this.cdpProxy.stopServer();
+    disposeAll(this.disposables);
+    vscode.commands.executeCommand("workbench.action.debug.stop", undefined, {
+      sessionId: this.session.id,
+    });
+  }
+
+  private async ping() {
+    try {
+      const result = await this.cdpProxy.injectDebuggerCommand({
+        method: "Runtime.evaluate",
+        params: {
+          expression: "('ping')",
+        },
+      });
+      if ("value" in result && result.value === "ping") {
+        this.sendEvent(new Event("RNIDE_pong"));
+      }
+    } catch (_) {
+      /** debugSession is waiting for an event, if it won't get any it will fail after timeout, so we don't need to do anything here */
+    }
+  }
+
+  private async startProfiling() {
+    await this.cdpProxy.injectDebuggerCommand({ method: "Profiler.start", params: {} });
```

Comment:
I wonder if we could use profiling command from vscode-js-debug now that we use it directly: https://github.com/microsoft/vscode-js-debug/blob/main/src/ui/profiling/uiProfileManager.ts#L113

Reply:
I looked into it briefly, and it seemed from the CDP messages sent between the debugger and the app that js-debug doesn't think the debuggee supports profiling. I didn't spend any time beyond that on it though, so there might be a way to make it work.
I'd keep it out of this PR though.